mesonbuild/scripts/run_tool.py

```diff
@@ -127,3 +129,10 @@ def run_clang_tool(name: str, srcdir: Path, builddir: Path, fn: T.Callable[...,
     def wrapper(path: Path) -> T.Iterable[T.Coroutine[None, None, int]]:
         yield fn(path, *args)
     return asyncio.run(_run_workers(all_clike_files(name, srcdir, builddir), wrapper))
+
+def run_tool_on_targets(fn: T.Callable[[T.Dict[str, T.Any]],
+                                       T.Iterable[T.Coroutine[T.Any, T.Any, int]]]) -> int:
+    targets = load_info_file(get_infodir(os.getcwd()), kind='targets')
```

Comment:
I don't think we need to import mintro and its attendant module dependency tree *just* so we can programmatically get filenames which are documented as a stable API in https://mesonbuild.com/IDE-integration.html